### DISCUSSION TITLE
Vercel Blob Cost/Price

### DISCUSSION THREAD

**Discussion by zidkim** [vercel/community#2836] [Help]
### Summary

I understand that this product is in private beta and things may change. 
That said, I'd still like to inquire about this from a cost perspective as I need to select a solution soon.

In short, I'm trying to assess the options between:
• Backblaze + Cloudflare
• Vercel Blob

The only thing I don't like about B+C is that it introduces a single point of failure where i have to refresh authorization token to ensure private access to images.

Needless to say, the B+C combo is much cheaper as there are no egress costs, but Vercel's caching/edge architecture helps a lot with this as well, at least for static images. I'm going to launch a social media app towards the end of the year and for simplicity sake, say 80% of images are static and 20% are dynamic, at least 1x a day dynamic.

Let's assume that due to caching, only 20% of image requests result in actual data egress from the storage service (i.e., 80% of the requests are served from the cache). Not sure about the actual cache hit ratio as it could be higher or lower depending on various factors such as user behavior, size of image library, cache eviction policy, etc.

So, if you have 100,000 users and each user requests 1000 images per month, that's a total of 100,000,000 image requests per month. Considering each image is approximately 190KB, without caching, this would result in approximately 19,000,000,000KB, or around 18,120GB (or ~17.7TB) of egress data.

With the 20% caching assumption, the egress data would be reduced to around 3,624GB (or ~3.5TB).
For Vercel, the first 1GB of egress per month is free. The additional egress cost is $0.15 per GB. Subtracting the free 1GB from our total of 3,624GB, we get 3,623GB of billable egress. Multiplying by the cost, the total egress cost would be around $543.45.

Is this a fair assessment?
Storage is not my playground but from a quick glance, the cost seems to be the highest in the industry?
Thank you in advance. I know the product is still in beta so things may change and Im not  trying to complain, mainly determine if this is a viable route for my project and others in the future

### Steps to Reproduce

No steps to reproduce

**Comment by smaeda-ks** [comment]
duplicate of https://github.com/orgs/vercel/discussions/2837
